File: MPI/mpicheck/tests/functions/2.0/MPI_Type_create_hindexed.f90

        
        program main
        use mpi
        
        INTEGER var_0
       INTEGER var_1(10)
       INTEGER(KIND=MPI_ADDRESS_KIND) var_2(10)
       INTEGER var_3
       INTEGER var_4
       INTEGER var_5
        call mpi_type_create_hindexed(var_0, var_1, var_2, var_3, var_4, var_5)
       call pmpi_type_create_hindexed(var_0, var_1, var_2, var_3, var_4, var_5)
        end program main
    

File: MPI/mpicheck/tests/functions/2.0/MPI_Type_create_hindexed.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    int var_0;
    const int var_1[2];
    const MPI_Aint var_2[2];
    MPI_Datatype var_3;
    MPI_Datatype *var_4;
    int ret;
    /* calls */
    ret = MPI_Type_create_hindexed(var_0, var_1, var_2, var_3, var_4);
    ret = PMPI_Type_create_hindexed(var_0, var_1, var_2, var_3, var_4);
    return 0;
}
